README.md

```diff
@@ -1,2 +1,288 @@
+
+    ![](https://i.imgur.com/1IjtgOg.png)
+
+    In above Image, it is the General Equation of Length (vertical distance between the axis of the hip joints and the foot joint)
+    Here, G (Gear Ratio) which is taken to be 1 for our Bot.
+    
+    
+    ![](https://i.imgur.com/9046wKp.png)
+
+    The above Graph gives is about change of Leg Extension wrt change in Driver Motor Angle. 
+
+
+    [Link To Python Script Used]()
+    [LInk to Python Script to Plot the Graph]()
+
+* **Specfications of Bot**
+    <font color="red">
+    
+    Lmax = 13 cm
+
+    Lmin = 4 cm
+    
+    L1 = 4.286 cm
+    
+    L2 = 9.236 cm
+
+    L3 = 3.5 cm
+
+    Phi1 = 45
+
+    Phi2 = 15
+
+    </font>
+
+
+
+
+## <ins> Components
+* Bot Parts of Acrylic Material (Laser-Cutted)
+* N20 motor Shaft for Wheels (3D Printed)
+* SRA PCB 
+* N20 motors
+* MG90 Servo Motors
+* Buck Converter (LM2596)
+* Bearings
+* Shafts
+
+> :memo: **Note:** [Link to Components used In Robot and Description](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/Components.md)
+
+## <ins> PID
+A PID controller is one kind of device used to control different process variables like pressure, flow, temperature, and speed in industrial applications. 
+
+> :memo: **Note:** [PID Notes](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/PID%20Explanation)
+
+## <ins> Code Flow
+
+Here, we First wrote the algorithm for Self-Balancing the bot.
+
+
+For self-balancing the bot, we need to keep the bot a bit above our desired angle (Setpoint), since the bot is very fast and will fall down once it's past the desired angle.
+
+The pitch angle is required for Self-Balancing which is obtained from the Inertial Measurement Unit(IMU sensor) plug-in.
+
+> :memo: **Note:**[Link to Explanation of Self Balancing Code](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/Self%20Balancing%20Code.md)
+
+
+And after Successful Run of Self Balancing Code, we than
+wrote the code for Servo moving from Maximum to Minimum Length (Variable Height).
+
+[Link to Code]()
+
+And after that we wrote the code of integrating Self Balancing Code with servos locking up at minimum height for Self Balancing of Bot at Minimum Height.
+
+[Link to Code](https://github.com/KudoC0nan/Balance_On_Wheels/tree/dev_arsh/Self_Balancing_Servo_set_zero)
+
+**Flow Chart of Code**
+![](https://i.imgur.com/6f7omtf.png)
+
+After Which we wrote the Jumping Code for Bot to Jump from Minimum Position.
+
+[Link to Code]()
+
+After Which we wrote the code for integration of Jumping Code with Self Balancing Code.
+
+[Link to Code]()
+
+---
+## <ins>Results and Demo
+
+--- 
+## <ins>Future Work
+1) Segway Bot , in which we can add camera ,speech recognition and self driving facility in a confined space. We can also add arms to it . So this Bot can interact with people (by recognizing their face and voice) ,can change height and  with arms can lift up objects.(Assistant Type)   
```

Comment:
Everything need not be explained here just pointers explaining future work will do and what you have mentioned are mainly ideas not algorithms per se which is fine but explain very short crisp points. Last Point is most doable keep that first, like implementing position control using LQR controller, roll stabilization are things you'd need to do implement that point First two only short mentions.

Reply:
Review to these future applications was already given on discord. So , don't add anything just to add ...

Reply:
Shortened applications and made it more specific

Reply:
All three are applications except for LQR implementation I dont know whether they count as future work or Applications per se, so I  personally dont feel if they should be include here. LQR and Roll Stablization se you can move on uneven terrain. Please Research about LQR controller its not just for Positional Control, but rather you can include position control with self balancing using LQR of course for that you will require motor with enocder. If not understood context of Roll Stabilzation then you should have asked me. Dont just put ideas here for sake of it. @Arsh-Khan @KudoC0nan

Reply:
will update!